### ISSUE TITLE
WhatsApp Notification count not working.

### ISSUE THREAD

**Issue by FingerlessGlov3s** [getferdi/ferdi#818]
**Bug description:**
WhatsApp notification count not working on the sidebar, I do get notifications though.

**Steps to reproduce:**
Add WhatsApp and then get someone to message you.

**Expected behavior**
Notification count to show.

**Screenshots**
![image](https://user-images.githubusercontent.com/3495647/84577962-b1fda800-adb8-11ea-813b-6c9a718f2b7a.png)

**Environment:**

- Operating System: Windows 10 1909
- Ferdi Version: 5.5.0 (x64)
- Server: Ferdi Self Hosted

**Comment by FingerlessGlov3s** [comment]
Oh also the notification comes through without any profile picture or whatsapp Icon. Not sure which its suppose to have.
![image](https://user-images.githubusercontent.com/3495647/84596792-a90fe380-ae57-11ea-952b-178ebb03537e.png)

**Comment by crifonju** [comment]
This is also happening to me. 
MacOS High Sierra 10.13.6
Ferdi 5.5.0 (5.5.0.2116)

**Comment by iliyang** [comment]
Same here:
MacOS Catalina 10.15.1
Version 5.5.0 (5.5.0.2116)

**Comment by VOOM108** [comment]
Win10 5.5.0 as well

**Comment by mahadevans87** [comment]
A fix has been put up here - https://github.com/getferdi/recipes/pull/206. Till it gets merged, go to your whatsapp recipe's `webview.js` and change the following line of code - 
`const elements = document.querySelectorAll("[class=OUeyt]");` 
to
`const elements = document.querySelectorAll("[class=_31gEB]");`

**Comment by patrickrushton** [comment]
Same happening here 5.5.0 macOS Mojave. WhatsApp notifications come through but no badge appears in the Ferdi sidebar.

**Comment by FingerlessGlov3s** [comment]
@patrickrushton 

See post by mahadevans87, that fixes it but I've also noticed the WhatsApp profile photos don't show either when the notification comes through.

**Comment by billnguyense** [comment]
Seems that the problem comes back.
I have just added WhatsApp to MacOS Catalina and the notification count is still not showing

**Comment by Grosbelli** [comment]
I have the same problem

**Comment by Lucria** [comment]
Hey all, so it wasn't working for me initially (am using Windows 10 Version 2004), but I decided to use Franz's implementation for their Whatsapp recipe in `webview.js`. So similarly, replace 
`const elements = document.querySelectorAll("[class=OUeyt]");`
with 
`const elements = document.querySelectorAll('.CxUIE, .unread, ._0LqQ, .m61XR .ZKn2B ');`

**Comment by veresvit** [comment]
Hi, another update from Franz repo (works as of today):

`const elements = document.querySelectorAll('.CxUIE, .unread, ._0LqQ, .m61XR .ZKn2B, .VOr2j');`

Source:
https://github.com/meetfranz/recipe-whatsapp/blob/master/webview.js#L38

**Comment by billnguyense** [comment]
This works for me. Thanks a alot!
> Hi, another update from Franz repo (works as of today):
> 
> `const elements = document.querySelectorAll('.CxUIE, .unread, ._0LqQ, .m61XR .ZKn2B, .VOr2j');`
> 
> Source:
> https://github.com/meetfranz/recipe-whatsapp/blob/master/webview.js#L38

**Comment by JPnux** [comment]
hi
same problem with 5.6.0 beta-5 version on Linux Ubuntu

works with "veresvit" post :-) thank you

**Comment by woernsn** [comment]
Out of curiosity (because after a few months I finally decided to look into the same problem for me):
How often are these fixes pulled from Franz?

**Comment by mauro-miatello** [comment]
same here, Mint 20.1

**Comment by crilofer** [comment]
Same here, Ubuntu 20.04.1 LTS. Ferdi 5.5.0

**Comment by manavortex** [comment]
Current fix (worked for me)
`const elements = document.querySelectorAll('.CxUIE, .unread, ._0LqQ, .m61XR .ZKn2B, .VOr2j, span[aria-label*="unread"] ');`

**Comment by gustavoortega** [comment]
> Current fix (worked for me)
> `const elements = document.querySelectorAll('.CxUIE, .unread, ._0LqQ, .m61XR .ZKn2B, .VOr2j, span[aria-label*="unread"] ');`

I am using the latest beta version and this workaround was the only fix for me, thanks!

**Comment by hmalaud** [comment]
Wasn't working for me, had to add ", ._2TiQe" to the list of elements. I'm wondering if '[role="gridcell"] > span:not(:empty)' instead of that whole list wouldn't be better (it seems to be working fine, but it would need to be tested on a longer period of time.

**Comment by empz** [comment]
This is still happening in the latest release (which is a year old). Are you planning to release a new version already or what?

**Comment by vraravam** [comment]
@empz - As I had posted earlier in the discussion - have you upgraded to the nightlies? They are quite stable, and many users have found such problems have already been fixed. We are trying to contact the main owner (@kytwb ) to see if he can unblock the signing key for the windows-builds - this is the last piece that is holding us back from making a public release.

The latest nightlies can be found [here](http://github.com/getferdi/nightlies/releases)
